### PR TITLE
Use entrypoint from LC_UNIXTHREAD

### PR DESCRIPTION
**Pull request by mrmacete**
fixes https://github.com/radare/radare2/issues/13519

### CODE REVIEW

File: libr/bin/p/bin_xnu_kernelcache.c [hunk 1]
```diff
@@ -723,6 +723,11 @@ static RList *entries(RBinFile *bf) {
 	}
 
 	RKernelCacheObj *kobj = (RKernelCacheObj*) obj->bin_obj;
+	ut64 entry_vaddr = kobj->mach0->entry;
+	ut64 entry_paddr = entry_vaddr - kobj->pa2va_exec;
```
Comment (radare) [inline libr/bin/p/bin_xnu_kernelcache.c]: can we assume entry_vaddr will be always be bigger than pa2va_exec ? can we have  a check?

File: libr/bin/p/bin_xnu_kernelcache.c [hunk 2]
```diff
@@ -723,6 +723,11 @@ static RList *entries(RBinFile *bf) {
 	}
 
 	RKernelCacheObj *kobj = (RKernelCacheObj*) obj->bin_obj;
+	ut64 entry_vaddr = kobj->mach0->entry;
```
Comment (radare) [inline libr/bin/p/bin_xnu_kernelcache.c]: kobj can be null?

File: libr/bin/p/bin_xnu_kernelcache.c [hunk 3]
```diff
@@ -723,6 +723,11 @@ static RList *entries(RBinFile *bf) {
...
+	ut64 entry_paddr = entry_vaddr - kobj->pa2va_exec;
+	RBinAddr *ba = newEntry (entry_paddr, entry_vaddr, 0);
+	r_list_append (ret, ba);
```
Comment (radare) [inline libr/bin/p/bin_xnu_kernelcache.c]: check for null if newEntry fails

File: libr/bin/p/bin_xnu_kernelcache.c (hunk 2 above)
Comment (mrmacete) [reply libr/bin/p/bin_xnu_kernelcache.c]: well, there's already a check right before

**Pull request merged**